core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/index/SolrIndexServiceImpl.java

```diff
@@ -183,11 +187,13 @@ public void buildIndex() throws IOException, ServiceException {
 
     @Override
     public void postBuildIndex() throws IOException, ServiceException {
-        // this is required to be at the very very very end after rebuilding the whole index
-        optimizeIndex(solrConfiguration.getReindexCollectionName(), solrConfiguration.getReindexServer());
-        // Swap the active and the reindex cores
-        if (!solrConfiguration.isSingleCoreMode()) {
-            shs.swapActiveCores(solrConfiguration);
+        if(optimizeEnabled) {
+            // this is required to be at the very very very end after rebuilding the whole index
+            optimizeIndex(solrConfiguration.getReindexCollectionName(), solrConfiguration.getReindexServer());
+            // Swap the active and the reindex cores
+            if (!solrConfiguration.isSingleCoreMode()) {
```

Comment:
Should the core swap be inside the optimizeEnabled block?  Seems like the core should be swapped regardless if optimize is enabled.

Reply:
ah, yes, forgot that we when reindex we add to "reindex" core and swap should happen anyway